packages/ERTP/src/paymentLedger.js

```diff
@@ -458,56 +380,127 @@ export const vivifyPaymentLedger = (
     issuerBaggage,
     name,
     assetKind,
-    getBrand,
+    brand,
+    PurseIKit,
     harden({
       depositInternal,
       withdrawInternal,
     }),
   );
 
-  const issuer = vivifySingleton(issuerBaggage, `${name} issuer`, {
-    getBrand: () => brand,
-    getAllegedName: () => name,
-    getAssetKind: () => assetKind,
-    getDisplayInfo: () => displayInfo,
-    makeEmptyPurse,
-
-    isLive,
-    getAmountOf,
-    burn,
-    claim,
-    combine,
-    split,
-    splitMany,
-  });
+  /** @type {Issuer} */
+  const issuer = vivifyFarInstance(issuerBaggage, `${name} issuer`, IssuerI, {
+    getBrand() {
+      return brand;
+    },
+    getAllegedName() {
+      return name;
+    },
+    getAssetKind() {
+      return assetKind;
+    },
+    getDisplayInfo() {
+      return displayInfo;
+    },
+    makeEmptyPurse() {
+      return makeEmptyPurse();
+    },
+    isLive(payment) {
+      // IssuerI delays calling this method until `payment` is a Remotable
+      return paymentLedger.has(payment);
+    },
+    getAmountOf(payment) {
+      // IssuerI delays calling this method until `payment` is a Remotable
+      assertLivePayment(payment);
+      return paymentLedger.get(payment);
+    },
 
-  const mint = vivifySingleton(issuerBaggage, `${name} mint`, {
-    getIssuer: () => issuer,
-    mintPayment,
+    burn(payment, optAmountShape = undefined) {
+      // IssuerI delays calling this method until `payment` is a Remotable
+      assertLivePayment(payment);
+      const paymentBalance = paymentLedger.get(payment);
+      assertAmountConsistent(paymentBalance, optAmountShape);
+      try {
+        // COMMIT POINT.
+        deletePayment(payment);
+      } catch (err) {
+        shutdownLedgerWithFailure(err);
+        throw err;
+      }
+      return paymentBalance;
+    },
+    claim(srcPayment, optAmountShape = undefined) {
+      // IssuerI delays calling this method until `srcPayment` is a Remotable
+      assertLivePayment(srcPayment);
+      const srcPaymentBalance = paymentLedger.get(srcPayment);
+      assertAmountConsistent(srcPaymentBalance, optAmountShape);
+      // Note COMMIT POINT within moveAssets.
+      const [payment] = moveAssets(
+        harden([srcPayment]),
+        harden([srcPaymentBalance]),
+      );
+      return payment;
+    },
+    combine(fromPaymentsPArray, optTotalAmount = undefined) {
+      // IssuerI does *not* delay calling `combine`, but rather leaves it
+      // to `combine` to delay further processing until all the elements of
+      // `fromPaymentsPArray` have fulfilled.
```

Comment:
> I assume this stems from inability to express the need for awaiting all elements of an array (as opposed to awaiting a single value).

Correct.

> It will be interesting to consider the use of decorators for a similar purpose if they get added to the language.

Do you think people (maybe us?) will build similar callWhen/await delaying abstractions using decorators?

Resolving.